fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricBlockLootTablesProvider.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.provider;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import com.google.common.collect.Sets;
+
+import net.minecraft.data.server.BlockLootTableGenerator;
+import net.minecraft.loot.LootTable;
+import net.minecraft.loot.LootTables;
+import net.minecraft.loot.context.LootContextType;
+import net.minecraft.loot.context.LootContextTypes;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
+import net.fabricmc.fabric.impl.datagen.FabricLootTableProvider;
+
+/**
+ * Extend this class and implement {@link FabricBlockLootTablesProvider#generateBlockLootTables}.
+ *
+ * <p>Register an instance of the class with {@link FabricDataGenerator#addProvider} in a {@link net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint}
+ */
+public abstract class FabricBlockLootTablesProvider extends BlockLootTableGenerator implements FabricLootTableProvider {
+	protected final FabricDataGenerator dataGenerator;
+
+	protected FabricBlockLootTablesProvider(FabricDataGenerator dataGenerator) {
+		this.dataGenerator = dataGenerator;
+	}
+
+	/**
+	 * Use the range of {@link BlockLootTableGenerator#addDrop} methods to generate block drops.
+	 */
+	protected abstract void generateBlockLootTables();
+
+	@Override
+	public LootContextType getLootContextType() {
+		return LootContextTypes.BLOCK;
+	}
+
+	@Override
+	public FabricDataGenerator getFabricDataGenerator() {
+		return dataGenerator;
+	}
+
+	@Override
+	public void accept(BiConsumer<Identifier, LootTable.Builder> biConsumer) {
+		generateBlockLootTables();
+
+		Set<Identifier> generated = Sets.newHashSet();
+
+		for (Map.Entry<Identifier, LootTable.Builder> entry : lootTables.entrySet()) {
+			Identifier identifier = entry.getKey();
+
+			generated.add(identifier);
+
+			if (identifier.equals(LootTables.EMPTY)) {
+				continue;
+			}
+
+			biConsumer.accept(entry.getKey(), entry.getValue());
+		}
+
+		if (dataGenerator.isStrictValidationEnabled()) {
```

Comment:
Not sure as to the cause atm but this doesn't seem to work for expanded storage, even when all loot table code is there it complains every block doesn't have a loot table...
Code: <https://github.com/NinjaPhenix/ExpandedStorage/blob/7.3/1.18/fabric-datagen/src/main/java/ninjaphenix/expandedstorage/data/providers/BlockLootProvider.java>

Error:
```
[17:45:55] [Render thread/INFO] (Minecraft) [STDERR]: java.lang.IllegalStateException: Missing loot table(s) for [expandedstorage:netherite_barrel, expandedstorage:obsidian_chest, expandedstorage:obsidian_barrel, expandedstorage:old_wood_chest, expandedstorage:candy_cane_mini_present, expandedstorage:diamond_barrel, expandedstorage:wood_mini_chest, expandedstorage:old_diamond_chest, expandedstorage:present, expandedstorage:red_mini_present_with_sparrow, expandedstorage:diamond_chest, expandedstorage:iron_chest, expandedstorage:netherite_chest, expandedstorage:vanilla_wood_mini_chest, expandedstorage:wood_mini_chest_with_sparrow, expandedstorage:old_obsidian_chest, expandedstorage:wood_chest, expandedstorage:white_mini_present, expandedstorage:red_mini_present, expandedstorage:pumpkin_mini_chest, expandedstorage:pink_amethyst_mini_present, expandedstorage:old_iron_chest, expandedstorage:iron_barrel, expandedstorage:vanilla_wood_mini_chest_with_sparrow, expandedstorage:green_mini_present_with_sparrow, expandedstorage:old_netherite_chest, expandedstorage:gold_barrel, expandedstorage:pumpkin_mini_chest_with_sparrow, expandedstorage:gold_chest, expandedstorage:pink_amethyst_mini_present_with_sparrow, expandedstorage:lavender_mini_present, expandedstorage:white_mini_present_with_sparrow, expandedstorage:candy_cane_mini_present_with_sparrow, expandedstorage:green_mini_present, expandedstorage:lavender_mini_present_with_sparrow, expandedstorage:pumpkin_chest, expandedstorage:old_gold_chest]
```

Reply:
Fixed. Thanks 👍